buildvertical.targets

```diff
@@ -27,13 +27,21 @@
     </MSBuild>
 
     <ItemGroup>
-      <!-- transform back to project -->
-      <ProjectWithConfiguration Include="@(_ProjectBestConfigurations->'%(OriginalItemSpec)')">
-        <AdditionalProperties>Configuration=%(Identity);%(_ProjectBestConfigurations.AdditionalProperties)</AdditionalProperties>
+      <!-- transform back to project keeping the original item order. -->
+      <_ProjectBestConfigurations>
+        <Configuration>%(Identity)</Configuration>
+      </_ProjectBestConfigurations>
+      <ProjectWithConfiguration Include="@(_ProjectBestConfigurations->'%(OriginalItemSpec)')" />
+      <ProjectWithConfiguration>
+        <AdditionalProperties>Configuration=%(Configuration);%(ProjectWithConfiguration.AdditionalProperties)</AdditionalProperties>
       </ProjectWithConfiguration>
     </ItemGroup>
 
     <ItemGroup>
+      <ProjectWithConfiguration RemoveMetadata="Configuration" />
```

Comment:
What is RemoveMetadata? I've never seen that before. However if it is doing what I think it is that sees like some leftover thing that should be removed because we don't want to remove the Configuration.